tensorflow/compiler/jit/mark_for_compilation_pass.cc

```diff
@@ -1010,6 +1010,45 @@ Status MarkForCompilationPassImpl::BuildInitialClusterSet() {
   return Status::OK();
 }
 
+StatusOr<bool> IsIdentityDrivingConstsInLoop(Node* node) {
+  if (!node->IsIdentity()) {
+    return false;
+  }
+
+  // Check if one of the ancestors is a Switch node.
+  Node* switch_node = nullptr;
+  for (const Edge* e : node->in_edges()) {
+    if (e->src()->IsSwitch()) {
+      switch_node = e->src();
+      break;
+    }
+  }
+  if (switch_node == nullptr) {
+    return false;
+  }
+
+  // Check if the Switch is driven by LoopCond.
+  const Node* maybe_loopcond;
+  TF_RETURN_IF_ERROR(switch_node->input_node(1, &maybe_loopcond));
+  if (!maybe_loopcond->IsLoopCond()) {
+    return false;
+  }
+
+  // Check if the Identity is driving any const nodes through a control edge.
+  bool driving_any_consts = false;
+  for (const Edge* e : node->out_edges()) {
```

Comment:
Up to you, but you could use `absl::c_any_of` here.

Reply:
Done. It is indeed more neat with `absl::c_any_of`.